### PR TITLE
Refine error message in memory folder

### PR DESCRIPTION
**Pull request by zhiqiu**
### PR types

Others
### PR changes

Others
### Describe

Refine error message in memory folder

### CODE REVIEW

File: paddle/fluid/memory/allocation/cpu_allocator.cc [hunk 1]
```diff
@@ -37,8 +39,11 @@ Allocation *CPUAllocator::AllocateImpl(size_t size) {
 #ifdef _WIN32
   p = _aligned_malloc(size, kAlignment);
 #else
-  PADDLE_ENFORCE_EQ(posix_memalign(&p, kAlignment, size), 0, "Alloc %ld error!",
-                    size);
+  int error = posix_memalign(&p, kAlignment, size);
+  PADDLE_ENFORCE_EQ(
+      error, 0,
+      platform::errors::ResourceExhausted(
+          "Alloc memory of %ld size failed, error code is!", size, error));
```
Comment (chenwhql) [inline paddle/fluid/memory/allocation/cpu_allocator.cc]: `error code is!` -> `error code is %d.`?

File: paddle/fluid/memory/detail/system_allocator.cc [hunk 2]
```diff
@@ -55,10 +56,15 @@ void* AlignedMalloc(size_t size) {
 #ifdef _WIN32
   p = _aligned_malloc(size, alignment);
 #else
-  PADDLE_ENFORCE_EQ(posix_memalign(&p, alignment, size), 0, "Alloc %ld error!",
-                    size);
+  int error = posix_memalign(&p, alignment, size);
+  PADDLE_ENFORCE_EQ(
+      error, 0,
+      platform::errors::ResourceExhausted(
+          "Alloc memory of %ld size failed, error code is!", size, error));
```
Comment (chenwhql) [inline paddle/fluid/memory/detail/system_allocator.cc]: save above

File: paddle/fluid/memory/detail/system_allocator.cc [hunk 3]
```diff
@@ -55,10 +56,15 @@ void* AlignedMalloc(size_t size) {
...
 #endif
-  PADDLE_ENFORCE_NOT_NULL(p, "Fail to allocate CPU memory: size = %d .", size);
+  PADDLE_ENFORCE_NOT_NULL(
+      p, platform::errors::ResourceExhausted(
+             "Fail to allocate CPU memory: size = %d .", size));
```
Comment (chenwhql) [inline paddle/fluid/memory/detail/system_allocator.cc]: `size = %d .`, no need ` ` between `%d` and `.`

File: paddle/fluid/memory/detail/memory_block.cc [hunk 4]
```diff
@@ -38,7 +39,10 @@ MemoryBlock* MemoryBlock::GetRightBuddy(MetadataCache* cache) {
 void MemoryBlock::Split(MetadataCache* cache, size_t size) {
   auto desc = cache->LoadDesc(this);
   // make sure the split fits
-  PADDLE_ENFORCE_GE(desc->total_size, size);
+  PADDLE_ENFORCE_GE(desc->total_size, size,
+                    platform::errors::InvalidArgument(
+                        "The size of memory block to split is "
```
Comment (chenwhql) [inline paddle/fluid/memory/detail/memory_block.cc]: `The size of memory (%d) to free exceeds the size of allocated cuda pinned memory (%d)`, is it possible to use this form to inform the specific size?

File: paddle/fluid/memory/allocation/cpu_allocator.cc (hunk 1 above)
Comment (zhiqiu) [reply paddle/fluid/memory/allocation/cpu_allocator.cc]: Done.

File: paddle/fluid/memory/detail/system_allocator.cc (hunk 2 above)
Comment (zhiqiu) [reply paddle/fluid/memory/detail/system_allocator.cc]: Done, thx.

File: paddle/fluid/memory/detail/system_allocator.cc (hunk 3 above)
Comment (zhiqiu) [reply paddle/fluid/memory/detail/system_allocator.cc]: Done, thx.

**Review by chenwhql — APPROVED** [review]
LGTM

**Pull request merged**